src/bdecode.cpp

```diff
@@ -479,7 +479,7 @@ namespace libtorrent
 		{
 			bdecode_token const& t = tokens[token];
 			TORRENT_ASSERT(t.type == bdecode_token::string);
-			int const size = m_root_tokens[token + 1].offset - t.offset - t.start_offset();
```

Comment:
also, could you try to make ``tokens`` const? my assumption is that it won't actually change in this function.

`bdecode_token const* const tokens = m_root_tokens;`